### ISSUE TITLE
Can you share the Model file for testing on my side?

### ISSUE THREAD

**Issue by JafferWilson** [abisee/pointer-generator#37]
Dear repository maker,
I would like to test the results on my side. Will it be possible for you to share the trained model of tensorflow on which you have tested the results? I have checked the output and found the results interesting. Kindly, let me know whether you can share the trained model, so that if I need to train the model I will start from the last stop, i.e. from your model. Do let me know.

**Comment by abisee** [comment]
Hi @JafferWilson, 

Due to some technical details (e.g. we changed the code a lot before releasing to make it cleaner), the checkpoint for our trained model reported in the paper does not work directly with the released code. We've been intending to figure out how to convert it into a format that will work with the released code so that we can release it, but we've been busy with some other things (sorry!). This is still on our to-do list.

In the meantime you can look at our full output on the entire test set (see the link in the README). Also, there are some people in the Github comment sections who seem to have successfully trained the model to get similar results to those reported in the paper. Perhaps they can give you their trained checkpoint.

**Comment by JafferWilson** [comment]
@abisee It will be great. I will try to check the others issues. Meanwhile if you have any reference of the person please do share with me so I can ask him/her. Please, it will be great help.

**Comment by abisee** [comment]
@JafferWilson it was @joy369 in [this discussion](https://github.com/abisee/pointer-generator/issues/12) who seems to have had success.

**Comment by JafferWilson** [comment]
Hey.. Thank you for your help. I am closing this issue.

**Comment by abisee** [comment]
Pretrained model now available -- see README.

**Comment by JafferWilson** [comment]
@abisee  I am getting the following error, while trying to run the decode mode for the trained model that you have share:
```
python run_summarization.py --mode=decode  --data_path=better.bin --vocab_path=finished_files/vocab --log_root=log/directory --exp_name=pretrained_model
INFO:tensorflow:Starting seq2seq_attention in decode mode...
max_size of vocab was specified as 50000; we now have 50000 words. Stopping reading.
Finished constructing vocabulary of 50000 total words. Last word added: chaudhary
INFO:tensorflow:Building graph...
INFO:tensorflow:Adding attention_decoder timestep 0 of 1
INFO:tensorflow:Time to build graph: 0 seconds
2017-08-17 10:32:13.481390: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-17 10:32:13.481413: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-17 10:32:13.481429: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
2017-08-17 10:32:13.481435: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX2 instructions, but these are available on your machine and could speed up CPU computations.
2017-08-17 10:32:13.481440: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use FMA instructions, but these are available on your machine and could speed up CPU computations.
INFO:tensorflow:Loading checkpoint log/directory/pretrained_model/train/model-238410
INFO:tensorflow:Restoring parameters from log/directory/pretrained_model/train/model-238410
2017-08-17 10:32:13.526779: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/decoder/attention_decoder/lstm_cell/kernel not found in checkpoint
2017-08-17 10:32:13.527149: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/decoder/attention_decoder/lstm_cell/bias not found in checkpoint
2017-08-17 10:32:13.529141: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/bw/lstm_cell/bias not found in checkpoint
2017-08-17 10:32:13.531495: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/fw/lstm_cell/bias not found in checkpoint
2017-08-17 10:32:13.531495: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/bw/lstm_cell/kernel not found in checkpoint
2017-08-17 10:32:13.533088: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/fw/lstm_cell/kernel not found in checkpoint
INFO:tensorflow:Failed to load checkpoint from log/directory/pretrained_model/train. Sleeping for 10 secs...
INFO:tensorflow:Loading checkpoint log/directory/pretrained_model/train/model-238410
INFO:tensorflow:Restoring parameters from log/directory/pretrained_model/train/model-238410
2017-08-17 10:32:23.631564: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/decoder/attention_decoder/lstm_cell/bias not found in checkpoint
2017-08-17 10:32:23.636247: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/decoder/attention_decoder/lstm_cell/kernel not found in checkpoint
2017-08-17 10:32:23.638560: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/bw/lstm_cell/kernel not found in checkpoint
2017-08-17 10:32:23.641101: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/bw/lstm_cell/bias not found in checkpoint
2017-08-17 10:32:23.641189: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/fw/lstm_cell/kernel not found in checkpoint
2017-08-17 10:32:23.642738: W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/fw/lstm_cell/bias not found in checkpoint
INFO:tensorflow:Failed to load checkpoint from log/directory/pretrained_model/train. Sleeping for 10 secs...


```
Please let me know if there is any suggestion.

**Comment by chandra-siri** [comment]
@abisee I'm also getting the same error as @JafferWilson  reported. 
W tensorflow/core/framework/op_kernel.cc:1158] Not found: Key seq2seq/encoder/bidirectional_rnn/fw/lstm_cell/kernel not found in checkpoint

**Comment by abisee** [comment]
Have you got the latest version of the code? What version of tensorflow are you running?

**Comment by chandra-siri** [comment]
Yes. I'm running on the latest version of the code. ( I did git pull) 
Tensorfow version 1.2.1
I'm running the following code: python2 run_summarization.py --mode=decode --data_path=/home/schandra/cnn-dailymail/finished_files/chunked/test_* --vocab_path=/home/schandra/cnn-dailymail/finished_files/vocab --log_root=/home/schandra/logs/ --exp_name=pretrained_model --max_enc_steps=400 --max_dec_steps=120 --coverage=1

and I'm getting the error messages as reported above.

**Comment by abisee** [comment]
I think this might be due to a Tensorflow version difference (I've been running on Tensorflow 1.0)

**Comment by abisee** [comment]
OK, I just tried this on both Tensorflow 1.0 and Tensorflow 1.2.1. I get the same error for TF1.2.1 but it works on TF1.0. 

It looks like TF1.0 names the variables one way (that's how they're saved in the checkpoint) but TF1.2.1 names them differently, so when TF1.2.1 tries to load the variables, certain keys don't match.

For now can you run on TF1.0?

**Comment by abisee** [comment]
@chandra-siri @JafferWilson 

The README now contains links to versions for both TF1.0 and TF1.2.1. The only difference is the naming of some variables.

I used [this script](https://gist.github.com/batzner/7c24802dd9c5e15870b4b56e22135c96) to change the names of the variables in the checkpoint. It's very simple, so if you have any further Tensorflow version naming problems, the script may be useful to you.

**Comment by JafferWilson** [comment]
@abisee Thank you for sharing this with me. I think that the issue is due to the tensorflow version only. Hence, I used your TF - 1.2.1 version model and it worked effectively,  Closing the issue.

**Comment by chandra-siri** [comment]
@abisee Thanks a lot for helping out.
